app/src/androidTest/java/org/dhis2/usescases/teidashboard/robot/TeiDashboardRobot.kt

```diff
@@ -300,4 +303,27 @@ class TeiDashboardRobot : BaseRobot() {
             event++
         }
     }
+
+    fun clickOnStageGroup(programStageName: String) {
+        onView(withId(R.id.tei_recycler))
+            .perform(actionOnItem<DashboardProgramViewHolder>(hasDescendant(withText(programStageName)), click()))
+    }
+
+    fun clickOnEventGroupByStage(eventName: String) {
+        onView(withId(R.id.tei_recycler))
+            .perform(actionOnItem<DashboardProgramViewHolder>(hasDescendant(allOf(withText(eventName), withId(R.id.event_name))), click()))
+    }
+
+    fun checkEventWasDeletedStageGroup(teiProgramStages: TEIProgramStagesUIModel) {
+        val firstProgramStage = teiProgramStages.programStage_first
+        val secondProgramStage = teiProgramStages.programStage_second
+        val thirdProgramStage = teiProgramStages.programStage_third
+
+        onView(withId(R.id.tei_recycler))
+            .check(matches(allOf(
+                hasItem(allOf(hasDescendant(withText(firstProgramStage.name)), hasDescendant(withText(firstProgramStage.events)))),
+                hasItem(allOf(hasDescendant(withText(secondProgramStage.name)), hasDescendant(withText(secondProgramStage.events)))),
+                hasItem(allOf(hasDescendant(withText(thirdProgramStage.name)), hasDescendant(withText(thirdProgramStage.events))))
```

Comment:
Hello @Sharmyn28 not sure what you're trying to accomplish here. 

Are you matching that a specific grouping stage has one less event? If this is the case you only need to check one _**hasItem**_ in order to make this method generic for every program/tei ? What do you think?